### ISSUE TITLE
The oci_identity_tag resource argument validator_type doesn't accept the "DEFAULT" value

### ISSUE THREAD

**Issue by GrzegorzHejman** [oracle/terraform-provider-oci#1753]
The argument validator_type of the oci_identity_tag resource doesn't accept the "DEFAULT" value (which should be a valid value). It only accepts the "ENUM" value. 

The oci-go-sdk specifies those values as valid values:

// Set of constants representing the allowable values for BaseTagDefinitionValidatorValidatorTypeEnum
const (
	BaseTagDefinitionValidatorValidatorTypeEnumvalue BaseTagDefinitionValidatorValidatorTypeEnum = "ENUM"
	BaseTagDefinitionValidatorValidatorTypeDefault   BaseTagDefinitionValidatorValidatorTypeEnum = "DEFAULT"
)

**Comment by Rohitmalakar1999** [comment]
yes the reason behind this issue the code block of **validator** only required when you need **ENUM** type of validator and for **DEFAULT** terraform resource block dont require **validator** block to be passed.

There are two solutions to resolve this issue(Choose any one):
1. Remove **validator** {} complete block from your code if you need DEFAULT (Static) type of tag only.

**OR**

2. If you are writing terraform code which should support both ENUM and DEFAULT type of tag then use dynamic block with the condition applied by validatorType variable value.

Please find the code below for your reference
.
`  dynamic "validator" {
    for_each = each.value.validatorType == "ENUM" ? [{validator_type = each.value.validatorType, values = each.value.values }] : []
    content {
      validator_type = each.value.validatorType
      values = each.value.values
    }
  }`
  
  Here at for_each i passed a condition for validatorType variable.  If your var.validatorType value is **ENUM** then dynamic block will pass **validator** block of code with required values (list of tag values) and if your var.validatorType value is other than  **ENUM** (Can say **DEFAULT**) then dynamic block wont pass **validator** block and create static tag.

**Comment by GrzegorzHejman** [comment]
Hi @Rohitmalakar1999,

Thanks for your comment. I'm aware that you can work around this issue. The second option is what I use now, but it would be nice to have the provider accept the DEFAULT value to make it consistent with the SDKs and APIs.
This conditional logic looks awkward and obscures the real intention - setting the validatorType to ENUM or DEFAULT.

**Comment by Rohitmalakar1999** [comment]
@GrzegorzHejman  You are right we should have both values supported instead of one type of value support. 
This case is the same for resource identity_tag_default as well where value is the required field.

The Oci team should update the resource and support both values

**Comment by tf-oci-pub** [comment]
Thank you for reporting the issue. We have raised an internal ticket to track this. Our service engineers will get back to you.

**Comment by GrzegorzHejman** [comment]
3,5 years reaction time. Impressive!
